services/controlplane/identity/identity.go

```diff
@@ -0,0 +1,61 @@
+package identity
+
+import "net/http"
+
+type Identifier interface {
```

Comment:
I know that names and concepts are subjective and always the hardest problem to solve, thus I am not insisting...

That being said, in my mental model what we are really abstracting here is control plane's different resource identifiers (urls). The authentication method always seems to be the same so far (basic with only a username).

So...
```
package controlplane

type HttpResource interface {
  URL() string
  BasicAuth() string
}

type workspaceResource struct {
  baseUrl string
  key string
  basicAuth string
}

func(nr *workspaceResource) URL() string {
 return path.Join(nr.baseUrl, "workspaces", nr.key)
}

type namespaceResource struct {
  baseUrl string
  key string
  basicAuth string
}

func(nr *namespaceResource) URL() string {
 return path.Join(nr.baseUrl, "namespaces", nr.key)
}
```

Finally, there seems to be some HTTP resource identifier inconsistency in control plane that needs to be resolved...

/data-plane/namespaces/:namespace/settings (plural without v1)
/data-plane/v1/namespaces/:namespace/config (plural with v1)
/data-plane/workspace/:workspaceId/settings (singular without v1)

Reply:
> abstracting here is control plane's different resource identifiers (urls). 

I get your point, but I had in my head a different kind of abstraction. 

Workspace and namespace are two ways the control plane can identify a rudder-server instance. In the feature, it might abstract things like auth scope and different auth mechanisms. The identifier will also be use for connecting using gRPC, for example. 

So the usage of Resource or Collection does not encapsulate what I have in mind for this.

I am happy to discuss this further, as I want Identifier to be used in communications with control-plane.

I hear the idea about `BasicAuth()`, although I don't see any problem with abstracting it from now.

Reply:
>Finally, there seems to be some HTTP resource identifier inconsistency in control plane that needs to be resolved...

I was waiting for a PR to merge in order to fix this

Reply:
>I get your point, but I had in my head a different kind of abstraction.

As long as the abstraction is properly documented I am fine